src/components/Main.js

```diff
@@ -1,29 +1,53 @@
 import {Component} from 'react';
 import HornedBeast from './HornedBeast.js';
-import {Container, Row, Col} from 'react-bootstrap';
-// import SelectedBeast from './SelectedBeast.js';
+import {Container, Row, Col, Form} from 'react-bootstrap';
 
 class Main extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      // showModal: false
+      beastsToShow: this.props.imageData,
     };
   }
 
+  handleChange = (event) => {
+    const selection = parseInt(event.target.value);
+    let filteredBeasts;
+
+    if (selection === 1) {
+      filteredBeasts = this.props.imageData.filter(beast => beast.horns === 1);
+    } else if (selection === 2) {
+      filteredBeasts = this.props.imageData.filter(beast => beast.horns === 2);
+    } else if (selection === 3) {
+      filteredBeasts = this.props.imageData.filter(beast => beast.horns === 3);
+    } else {
+      filteredBeasts = this.props.imageData.filter(beast => beast.horns === 100);
+    }
+
+    this.setState({beastsToShow: filteredBeasts});
+  }
```

Comment:
Hi Beau! Great work!

Your comment in canvas about how you previously attempted filter() was spot on! You had the right idea by using only 1 filter() and if the beast.horns === selection , then update the state. 

If you have time this weekend, maybe try to get it working with your original idea as you were close ! 👍